datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -145,6 +145,36 @@ impl TableProviderFactory for ListingTableFactory {
             },
         }?;
 
+        let create_local_path_mode = cmd
+            .options
+            .get("create_local_path")
```

Comment:
It will be great to eventually get the Options into a structured form (https://github.com/apache/arrow-datafusion/pull/7283#discussion_r1294557588) so we can have the compiler check values like this rather than relying on string matching. I have it on my list today to file a ticket about that